slicedimage/backends/_caching.py

```diff
@@ -18,7 +18,7 @@ class CachingBackend(Backend):
     def __init__(self, cacheroot, authoritative_backend, size_limit=SIZE_LIMIT):
         with CachingBackend._LOCK:
             if cacheroot not in CachingBackend._CACHE:
-                CachingBackend._CACHE[cacheroot] = Cache(cacheroot, size=int(size_limit))
+                CachingBackend._CACHE[cacheroot] = Cache(cacheroot, size_limit=int(size_limit))
```

Comment:
> re: Does reopening a cache with a smaller size clean up the directory?

Yes, re-opening the cache with a different size changes the setting. However, only so many objects are removed per call to `set()`. `cull()` can be called to reduce the size.

Reply:
Note: [constructor](https://github.com/grantjenks/python-diskcache/blob/master/diskcache/core.py#L362) takes `**settings`.